### PR TITLE
Example app

### PR DESCRIPTION
**Pull request by marcferna**
demo
screen shot 2018-05-19 at 8 00 55 pm
screen shot 2018-05-19 at 8 01 04 pm
screen shot 2018-05-19 at 8 01 10 pm

- App is built with Kotlin
- Have not added any tests

I've encounter a couple issues when using the library and that you might want to resolve (or I can check it out) before merging this.
1) When doing a custom dialog, the buttons didn't get the text that I set. This could be because I'm using the latest version of `material-dialogs`. (Screenshot 2)

2) When setting a custom click listener, it didn't trigger. I debugged it and it wasn't stopping in my breakpoint inside the listener... This might be also because of the `material-dialogs` version but I'm not sure. (Screenshot 3).

Happy to help resolve any of these.

### CODE REVIEW

File: example/app/src/main/java/com/apprate/example/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        // Custom dialog example
+        val customDialogBuilder = MaterialDialog.Builder(this)
+            .title(R.string.custom_title)
+            .content(R.string.custom_content)
+            .positiveText(R.string.custom_positive_text)
+            .neutralText(R.string.custom_neutral_text)
+            .negativeText(R.string.custom_negative_text)
+
+        findViewById<Button>(R.id.button_custom_dialog).setOnClickListener {
```
Comment (MSF-Jarvis) [inline example/app/src/main/java/com/apprate/example/MainActivity.kt]: Won't the use of AndroidKTX let you just do `buttonCustomDialog.setOnClickListener` here?

File: example/app/src/main/java/com/apprate/example/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            AppRate(this).setCustomDialog(customDialogBuilder).init()
+        }
+
+        // Custom click lister example
+        findViewById<Button>(R.id.button_custom_click_listener).setOnClickListener {
```
Comment (MSF-Jarvis) [inline example/app/src/main/java/com/apprate/example/MainActivity.kt]: Ditto

File: example/app/src/main/java/com/apprate/example/MainActivity.kt (hunk 1 above)
Comment (marcferna) [reply example/app/src/main/java/com/apprate/example/MainActivity.kt]: I'll try! First time using Kotlin! 😎

**Comment by MSF-Jarvis** [comment]
I'm on vacation beginning today so it'll be a few days before I can get back to this and look into why stuff is broken. Appreciate your contributions here!

**Comment by marcferna** [comment]
No worries! Let's check it out when you are back! 👍

File: example/app/src/main/java/com/apprate/example/MainActivity.kt (hunk 1 above)
Comment (MSF-Jarvis) [reply example/app/src/main/java/com/apprate/example/MainActivity.kt]: Awesome. Kotlin's pretty fun 😁

File: example/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    implementation 'com.android.support:appcompat-v7:27.1.1'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.0'
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    // Custom material dialogs
+    implementation 'com.afollestad.material-dialogs:core:0.9.6.0'
+
+    // Import AppRate
+    implementation 'com.github.MSF-Jarvis:AppRate:1.3'
```
Comment (MSF-Jarvis) [inline example/app/build.gradle]: Change this to `implementation project(':app')`, easier to test library changes this way ;)

**Comment by MSF-Jarvis** [comment]
After making all the changes I proposed, try upgrading the `material-dialogs` dependency in the library and see if it indeed is what's causing the bugs you're seeing.

**Comment by MSF-Jarvis** [comment]
I've done some work on this PR, looks like you didn't enable maintainer edits so I couldn't push it to your repository. My changes are in the `example-app` branch of *this* repo, please edit the PR to change the head to it. I've added you for collaboration so you can make further changes to that branch yourself.

**Comment by MSF-Jarvis** [comment]
FWIW the issues are not related to the material-dialogs version, more likely that I broke some API while rewriting in kotlin. I'll debug it after my vacation, SSH gets too hardcore for this.

**Pull request closed without merging**